libcontainer/cgroups/systemd/common.go

```diff
@@ -231,18 +230,21 @@ func systemdVersion(cm *dbusConnManager) int {
 	return version
 }
 
-func systemdVersionAtoi(verStr string) (int, error) {
-	// verStr should be of the form:
-	// "v245.4-1.fc32", "245", "v245-1.fc32", "245-1.fc32" (without quotes).
+func systemdVersionAtoi(str string) (int, error) {
+	// str should be of the form:
+	// "v245.4-1.fc32", "245", "v245-1.fc32", "245-1.fc32" (with quotes).
 	// The result for all of the above should be 245.
-	// Thus, we unconditionally remove the "v" prefix
+	// Thus, we unconditionally remove the "v prefix
 	// and then match on the first integer we can grab.
-	re := regexp.MustCompile(`v?([0-9]+)`)
-	matches := re.FindStringSubmatch(verStr)
-	if len(matches) < 2 {
-		return 0, fmt.Errorf("can't parse version %s: incorrect number of matches %v", verStr, matches)
+	str = strings.TrimLeft(str, `"v`)
```

Comment:
OH! Actually, this is not correct; this should be `strings.TrimPrefix`. `strings.TrimLeft` takes a "cut set" of characters, so currently `"vvvvvv""v"v"v""v245.4-1.fc32"` would be accepted;

https://go.dev/play/p/-Z4D9U4-oYM

Reply:
This is correct since it allows us to Trim either `v` or `"v` (i.e. with or without the quote).

Sure, it will also cut `v"` or `vvvv"""vvvv` but we do not expect such input here (the input expectations are documented).

Within the realm of expected input,
```go
strings.TrimLeft(str, `"v`)
```
work the same as
```go
strings.TrimPrefix(str, `"`)
strings.TrimPrefix(str, `v`)
```
but is somewhat more compact.

Reply:
@thaJeztah PTAL ^^